### ISSUE TITLE
Player Sync Event?

### ISSUE THREAD

**Issue by TheBrokenRail** [FabricMC/fabric#542]
This would be something like Cardinal Components' [```PlayerSyncCallback```](https://github.com/OnyxStudios/Cardinal-Components-API/blob/92fd50b4dbfa2adc51154c263e2d76961bbc94c0/cardinal-components-entity/src/main/java/nerdhub/cardinal/components/api/event/PlayerSyncCallback.java). This is needed because if you want to sync data to the client from ```ServerPlayerEntity``` to the client's ```PlayerEntity``` you either have to use Mixins in a lot of places to catch all sync events or sync every ```playerTick``` which is inefficient. It would be usefull to just be able to hook into a event.

**Comment by liach** [comment]
Yeah, this is a good event to indicate mods to sync player info to the client. On a side note, I wonder if these events should have more parameters indicating if all attributes should be sent or just changed attributes need to be synced (e.g. changes in dim may need a full send while others can just have partial)
